src/lib/dma.c

```diff
@@ -388,14 +385,13 @@ int dma_buffer_copy_from_no_consume(struct comp_buffer __sparse_cache *source,
 			   audio_stream_sample_bytes(istream);
 	uint32_t sink_bytes = audio_stream_sample_bytes(&sink->stream) *
 			      samples;
-	int ret;
 
 	/* process data */
-	ret = process(istream, 0, &sink->stream, 0, samples);
+	process(istream, 0, &sink->stream, 0, samples);
 
 	buffer_stream_writeback(sink, sink_bytes);
 
 	comp_update_buffer_produce(sink, sink_bytes);
 
-	return ret;
+	return 0;
```

Comment:
1. pcm converter function type is not unified, some are returning samples, some are void. 
    so, this patch mainly cover unify the return type for converter function.

2. regarding improve part, it is not big, since only saved two ret assignment.

main purpose is 1, 2 is extra benefit, there is no need to return a useless samples for a processing function.